include/swift/Reflection/TypeRefBuilder.h

```diff
@@ -572,9 +570,23 @@ class TypeRefBuilder {
         args.end() - argsIndex - numGenericArgs, args.end() - argsIndex);
 
     const BoundGenericTypeRef *parent = nullptr;
-    if (node->hasChildren())
-     parent = createBoundGenericTypeReconstructingParent(
-        node->getFirstChild(), decl, --shapeIndex, args, argsIndex + numGenericArgs);
+    if (node->hasChildren()) {
+      // Skip over nodes that are not of type class, enum or struct
+      auto parentNode = node->getFirstChild();
+      while (parentNode->getKind() != Node::Kind::Class &&
+             parentNode->getKind() != Node::Kind::Structure &&
+             parentNode->getKind() != Node::Kind::Enum) {
+        if (parentNode->hasChildren()) {
+          parentNode = parentNode->getFirstChild();
```

Comment:
Two questions:

1. Is it always the case that the nodes of interest will be the first child?
2. The recursion is indiscriminate about which nodes it passes over, are there any nodes it should stop recursing on?

Reply:
1. My understanding is that yes, it'll always be the first child. 

2. Yes. When I wrote the old code I assumed that any kind that had a generic counterpart was valid. As per my other comment, we found that the only ones that are handled (for generic typerefs) are structs classes and enums.